actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ShadowUserMigrationScheduler.java

```diff
@@ -0,0 +1,312 @@
+        }
+    }
+
+    private void updateBulkUserTable(Map<String, Object> propertiesMap) {
+        Response response = cassandraOperation.updateRecord(bulkUploadDbInfo.getKeySpace(), bulkUploadDbInfo.getTableName(), propertiesMap);
+        ProjectLogger.log("ShadowUserMigrationScheduler:updateBulkUserTable: status update result".concat(response.getResult() + ""), LoggerEnum.INFO.name());
+    }
+
+    private void updateMessageInBulkUserTable(String processId, String key, String value) {
+        Map<String, Object> propertiesMap = new HashMap<>();
+        propertiesMap.put(JsonKey.ID, processId);
+        propertiesMap.put(key, value);
+        if (value.equalsIgnoreCase(JsonKey.SUCCESS)) {
+            propertiesMap.put(JsonKey.STATUS, ProjectUtil.BulkProcessStatus.COMPLETED.getValue());
+            propertiesMap.put(JsonKey.FAILURE_RESULT, "");
+        } else {
+            propertiesMap.put(JsonKey.SUCCESS_RESULT, "");
+            propertiesMap.put(JsonKey.STATUS, ProjectUtil.BulkProcessStatus.FAILED.getValue());
+        }
+        updateBulkUserTable(propertiesMap);
+    }
+
+
+    /**
+     * this method will take descision wheather to update the record or not.
+     * @param shadowUser
+     * @param migrationUser
+     * @return boolean
+     */
+    private boolean isSame(ShadowUser shadowUser,MigrationUser migrationUser){
+        if(!shadowUser.getName().equalsIgnoreCase(migrationUser.getName())){
+        return false;
+        }
+        if(StringUtils.isNotBlank(migrationUser.getEmail()) && !shadowUser.getEmail().equalsIgnoreCase(migrationUser.getEmail())){
+            return false;
+        }
+        if(StringUtils.isNotBlank(migrationUser.getOrgExternalId()) && !shadowUser.getOrgExtId().equalsIgnoreCase(migrationUser.getOrgExternalId())){
+            return false;
+        }
+        if(StringUtils.isNotBlank(migrationUser.getPhone()) && !shadowUser.getPhone().equalsIgnoreCase(migrationUser.getPhone()))
+        {
+            return false;
+        }
+        if((getInputStatus(migrationUser.getInputStatus())!=shadowUser.getUserStatus())){
+            return false;
+        }
+
+        if(!migrationUser.getName().equalsIgnoreCase(shadowUser.getName())){
+            return false;
+        }
+        return true;
+    }
+
+    private boolean isOrgExternalIdValid(MigrationUser migrationUser) {
+        if (StringUtils.isBlank(migrationUser.getOrgExternalId())) {
+        return true;
+        }
+        Map<String, Object> request = new HashMap<>();
+        Map<String, Object> filters = new HashMap<>();
+        filters.put(JsonKey.EXTERNAL_ID, migrationUser.getOrgExternalId());
+        filters.put(JsonKey.CHANNEL, migrationUser.getChannel());
+        request.put(JsonKey.FILTERS, filters);
+        SearchDTO searchDTO = ElasticSearchHelper.createSearchDTO(request);
```

Comment:
This call for same orgExternalId, channel combination can be reduced by using local cahe

Reply:
@manzarul org externalId can be different also. or am I am missing something??

Reply:
Agree, My point was inside one org we can add N number of users, so for those n-1 we can cache and re-use

Reply:
@manzarul ya prepared a verifiedOrgExternalId set which will check if provided orgExternalId is present in set then return true else search eith respect to orgExternalId and channel and update the set if data found with provided input

Reply:
@anmol2302  we need to match orgExternalId and channel combination not only orgExternalId

Reply:
done